cmd/snap-confine/snap-confine.c

```diff
@@ -112,6 +112,10 @@ int main(int argc, char **argv)
 #endif				// ifdef HAVE_SECCOMP
 
 	if (geteuid() == 0) {
+		// ensure that "/" or "/snap" is mounted with the
```

Comment:
This is incorrect, you must do this while we hold a lock. Ideally the global lock in `/run/ns/.lock`; You want to add this to `sc_initialize_ns_groups` which already holds this lock and does similar treatment of `/run/snapd/ns`